test/parallel/test-zlib-sync-no-event.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+require('../common');
+const zlib = require('zlib');
+
+const zipper = new zlib.Gzip();
+zipper.on('close', () => { throw new Error('unexpected `close` event'); });
+
+const buffer = new Buffer('hello world');
+zipper._processChunk(buffer, zlib.Z_FINISH);
```

Comment:
You know, I think I like this more complete test regardless, so I'll push that up too.